src/ExplorerToggler.js

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import styled from "styled-components";
+
+import Explorer from "./Explorer";
+
+
+const TogglerContainer = styled.div`
+  margin: 20px 20px 0 20px;
+  text-align: center;
+`;
+
+class Toggler extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isHidden: true,
+      profile: '',  // Can be empty, `employee` or `employer`.
+    };
+  }
+
+  reset () {
+    this.setState({
+      isHidden: true,
+      profile: '',
+    });
+  }
+
+  setProfile (profile) {
+    if (profile && profile === this.state.profile) {
+      // Reset `profile` when the user click again on the same button.
+      profile = '';
+    }
+    this.setState({
+      profile: profile
+    });
+  }
+
+  toggleHidden () {
+    this.setState({
+      isHidden: !this.state.isHidden
+    });
```

Comment:
en théorie, quand on utilise le state courant dans un setState, il faut utiliser le mode "callback" : 


```js
this.setState(curState => ({
  isHidden: !curState.isHidden
}));
```

📚https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous